cfbs/git.py

```diff
@@ -1,7 +1,9 @@
 import os
 import tempfile
+from functools import partial
 from subprocess import check_call, check_output, run, PIPE, DEVNULL, CalledProcessError
 from cfbs.prompts import YES_NO_CHOICES, prompt_user
+from cfbs.cfbs_config import CFBSConfig
```

Comment:
I intentionally tried to keep the `CFBSConfig`-dependent (and thus cfbs-specific) code outside of this module to make it reusable. Especially the `commit_after_command` is very specific to the commands handling.

Reply:
It doesn't belong in commands and to me it makes the most sense to have all the git code in one place, but I can split it in two, `git.py` and `git_magic.py`.

Reply:
Added a new commits which makes `git.py` reusable and puts the rest of the code in `git_magic.py`.